### ISSUE TITLE
Build and test Bash without bazel

### ISSUE THREAD

**Issue by chaodaiG** [kubernetes/test-infra#23831]
**What would you like to be added**:

Build and test Bash without bazel, the related bazel rules are:

- sh_binary
- sh_test

Things to be considered:

- This will need to co-exist with current bazel rules for a short while, delete Bash test targets from bazel once this confirmed working
- Test driven by make rule and bash scripts
- Test results are in junit results format

**Why is this needed**:

This is part of #23796 

**Recommended testing approach**:
- Make test invokable by `make test`
- Create a PullRequest with change, manually run job by commenting `/test pull-test-infra-test-beta`
- [Optional]: if there is any tool that needs to exist in test image, update image at https://github.com/kubernetes/test-infra/blob/6e0d21ee50d690c8f78239257f588021d2ed57b5/config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml#L145

/sig testing
/help

**Comment by BenTheElder** [comment]
Worth pointing out: `sh_test` and `sh_binary` don't actually build anything, but they can be run in order to run a script, those rules pretty much just let you add shell into bazel's DAG, it doesn't do any linting or anything.

So really e.g.:
```bazel
sh_test(
    name = "verify-pylint",
    srcs = ["verify-pylint.sh"],
    data = [
        ":pylint_bin",
        "//:all-srcs",
    ],
    tags = ["lint"],
)
```

just needs to be part of #23830, the rest is covered by other issues like https://github.com/kubernetes/test-infra/issues/23897

I think we should be running shellcheck, but that doesn't appear to be part of CI, seems I only ever added it to other repos https://github.com/kubernetes/test-infra/issues/8090

**Comment by BenTheElder** [comment]
What doesn't seem to be covered by other issues I guess is verifying codegen.

**Comment by MadhavJivrajani** [comment]
Hey - sorry, I'm not able to find the bandwidth to work on this.
Unassigning myself for now, will try and take a look again at a later point if this isn't already picked up.
/unassign

**Comment by chaodaiG** [comment]
> What doesn't seem to be covered by other issues I guess is verifying codegen.

Verifying codegen was done by https://github.com/kubernetes/test-infra/pull/24252, so this issue can be closed once #23830 and #23897 are both done
